draft-ietf-jsonpath-base.md

```diff
@@ -289,50 +298,52 @@ JSONPath expressions are applied to a JSON value, the *argument*.
-The descendant operators (which start with `..`) select some or all of the descendants ({{descendant-selectors}}) of a node.
-The array slice
-syntax `[start:end:step]` allows selecting a regular selection of an
-element from an array, giving a start position, an end position, and
+An array slice `start:end:step` ({{slice}}) selects a series of
+elements from an array, giving a start position, an end position, and
 possibly a step value that moves the position from the start to the
-end ({{slice}}).
+end.
 
-Filter expressions are supported via the syntax `?(<boolean expr>)` as in
+Filter expressions `?<boolean expr>` select certain children of an object or array as in
 
 ~~~~
-$.store.book[?(@.price < 10)].title
+$.store.book[?@.price < 10].title
 ~~~~
 
 {{tbl-overview}} provides a quick overview of the JSONPath syntax elements.
 
-| JSONPath    | Description                                                                                                                         |
-|-------------|-------------------------------------------------------------------------------------------------------------------------------------|
-| `$`         | [the root node](#root-selector)                                                                                                     |
-| `@`         | the current node: within [filter selectors](#filter-selector)                                                                     |
-| `.name`     | child selectors for JSON objects: [dot selector](#dot-selector)                                                                     |
-| `['name']`  | child selectors for JSON objects: [index selector](#index-selector)                                                                 |
-| `..name` <br> `..[3]` | descendants: [descendant selector](#descendant-selectors)                                                                     |
-| `.*` <br> `[*]` | all child member values and array elements: [dot wildcard selector](#wildcard), [index wildcard selector](#index-wildcard-selector) |
-| `[3]`       | [index (subscript) selector](#index-selector): index current node as an array (from 0)                                              |
-| `[..,..]`   | [list selector](#list-selector): allow combining selector styles                                                                    |
-| `[0:100:5]` | [array slice selector](#slice): start:end:step                                                                                      |
-| `?...`      | [filter selector](#filter-selector)                                                                                                 |
-| `()`        | expression: within [filter selectors](#filter-selector)                                                                            |
+| JSONPath            | Description                                                                                                             |
```

Comment:
This should be sorted into query starts ('root identifiers'), query steps ('segments', sorted into the canonical ones and their shorthands), and selectors.
We need to make sure these are not mixed up in the minds of the readers.

Reply:
d4c8c39da0f1dc4c69fd1f9d5a6cb297a2e6c3e2 puts the identifiers first, then segments, then some selectors, then the shorthands. (Moving the shorthands up wouldn't seem to be very readable since they need to build on selectors.)

Reply:
Right.  Might be useful to have some section headers (we can do this in the next round).

Reply:
@cabo See https://github.com/ietf-wg-jsonpath/draft-ietf-jsonpath-base/pull/268 for a stab at improving the structure of the overview.